### DISCUSSION TITLE
The time precision in generated subtitle files is second, not millisecond

### DISCUSSION THREAD

**Discussion by whiteeat** [openai/whisper#809] [Q&A] [answered]
As the title says, the time precision in generated subtitle files is not milliseconds. So my question is, whether the code just work like that, or it is a bug? And, if I want the precision to be milliseconds, how can I achieve that?
![QQ截图20230105001125](https://user-images.githubusercontent.com/10330544/210602271-d4a4af98-fc85-4ce4-a4df-a87367573453.png)

**Comment by glangford** [comment, marked as answer]
From a comment by @jongwook in another discussion,

> Currently, the predicted timestamps tend to be biased towards integers, and there are some failure modes where the timestamps can be constantly shifted

See the discussion "Improving Timestamp Accuracy" in https://github.com/openai/whisper/discussions/435 where this is covered in detail.

For word level timestamps see https://github.com/openai/whisper/discussions/26 and this recently added notebook.
https://github.com/openai/whisper/blob/28769fcfe50755a817ab922a7bc83483159600a9/notebooks/Multilingual_ASR.ipynb

**Reply by whiteeat** [reply]
Thanks very much! I try solutions above and [stable-ts](https://github.com/jianfch/stable-ts) works best for me!

**Reply by kevinsaracho** [reply]
Como puedo solucionar el problema?

**Reply by glangford** [reply]
@kevinsaracho Use `--word_timestamps True`
